### PR TITLE
Add Zirgen-based acceleration

### PR DESCRIPTION
**Pull request by tzerrell**
Make use of the RSA acceleration from https://github.com/risc0/risc0/pull/2481 when run in the RISC Zero guest.

~Can't land until https://github.com/risc0/risc0/pull/2481 lands.~

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -64,3 +68,8 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [profile.dev]
 opt-level = 2
+
+# TODO: Point to a version tag once necessary features have stabilized
```
Comment (pdg744) [inline Cargo.toml:72]: If we're leaving this TODO in place, probably makes sense to open a linear ticket to help track it.

File: src/pkcs1v15/signature.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#[allow(unused_imports)]
```
Comment (flaub) [inline src/pkcs1v15/signature.rs:1]: Wondering why we need this?

File: Cargo.toml [hunk 3]
```diff
@@ -64,3 +68,8 @@ rustdoc-args = ["--cfg", "docsrs"]
...
+[patch.'https://github.com/risc0/risc0.git']
+risc0-circuit-bigint = { git = "https://github.com/risc0/risc0", branch = "main" }
+risc0-zkvm = { git = "https://github.com/risc0/risc0", branch = "main" }
```
Comment (flaub) [inline Cargo.toml]: Might be good to pin to a specific commit rather than using a branch ref.

File: src/pkcs1v15/signature.rs (hunk 2 above)
Comment (tzerrell) [reply src/pkcs1v15/signature.rs:1]: My hypothesis was that upstream was only warning on unused_imports whereas the risc0 repo has it set to error. But I'm not very familiar with how Rust warning/error settings work so I'm not sure if this sort of cross-crate setting contamination can even happen.

**Review by flaub — APPROVED** [review]

**Pull request merged**